python/ray/train/tests/test_backend.py

```diff
@@ -169,6 +213,32 @@ def train_func():
     assert set(e.finish_training()) == {0, 1}
 
 
+def test_local_world_size(ray_2_node_2_cpu):
+    config = TestConfig()
+    with patch.object(WorkerGroup, "add_workers", MockWorkerGroup.add_workers):
+        e = BackendExecutor(config, num_workers=4)
+        e.start()
+
+        def train_func():
+            return session.get_local_world_size()
+
+        e.start_training(train_func, dataset_spec=EMPTY_RAY_DATASET_SPEC)
+        assert set(e.finish_training()) == {2, 2, 2, 2}
```

Comment:
Btw, just to test a more complex case, I'm wondering if we should use 3 workers with expected results being `{2,1,2}`